manager/watcher/raw/raw.go

```diff
@@ -133,10 +133,12 @@ func (self *rawContainerWatcher) watchDirectory(dir string, containerName string
 	}
 	for _, entry := range entries {
 		if entry.IsDir() {
-			// TODO(vmarmol): We don't have to fail here, maybe we can recover and try to get as many registrations as we can.
-			_, err = self.watchDirectory(path.Join(dir, entry.Name()), path.Join(containerName, entry.Name()))
+			entryPath := path.Join(dir, entry.Name())
+			_, err = self.watchDirectory(entryPath, path.Join(containerName, entry.Name()))
```

Comment:
One example of potential issues is that kubelet hits the limits of the inotify watches, but it continues to start up without watching any pods. We can argue that this may happen anytime when kubelet is running, so there is no need to fail the startup. I am ok with that argument, but just want to point out that this is possible and we should try ensuring there is a recovery path. For example, maybe cadvisor should be not ready if errors like this keep happening, and kubelet should report node not ready?

I don't want to block this PR, but maybe we should open up an issue to discuss propagating the errors better. On the other hand, if cadvisor does **not** solely rely on inotify watches for notifications (e.g., it lists everything periodically), just ignore what I said :-)